In [1]:
import pandas as pd

## 1. Base nationale des IRVE (Infrastructures de Recharge pour Véhicules Électriques)

In [2]:
df_irve = pd.read_csv(
 'https://www.data.gouv.fr/api/1/datasets/r/eb76d20a-8501-400e-b336-d85724de5435'
)

print(f"Shape : {df_irve.shape}")
df_irve.head()

/tmp/ipykernel_1846/3761374698.py:1: DtypeWarning: Columns (12,18,19,20,21,22,24,25,29,33) have mixed types. Specify dtype option on import or set low_memory=False.
  df_irve = pd.read_csv(


Shape : (210129, 52)


,nom_amenageur,siren_amenageur,contact_amenageur,nom_operateur,contact_operateur,telephone_operateur,nom_enseigne,id_station_itinerance,id_station_local,nom_station,...,datagouv_resource_id,datagouv_organization_or_owner,created_at,consolidated_longitude,consolidated_latitude,consolidated_code_postal,consolidated_commune,consolidated_is_lon_lat_correct,consolidated_is_code_insee_verified,consolidated_is_code_insee_modified
0,ChargePoint,NaN,info@chargepoint.com,ChargePoint,info@chargepoint.com,NaN,ACU_Poste_De_Garde_Haguenau,ATHTBE1004017,ATHTBE1004017,ACU_Poste_De_Garde_Haguenau,...,b11113db-875d-41c7-8673-0cf8ad43e917,eco-movement,2023-06-28T11:46:08.539000+00:00,7.762694,48.825613,NaN,NaN,False,False,False
1,ChargePoint,NaN,info@chargepoint.com,ChargePoint,info@chargepoint.com,NaN,ACU_Poste_De_Garde_Haguenau,ATHTBE1004018,ATHTBE1004018,ACU_Poste_De_Garde_Haguenau,...,b11113db-875d-41c7-8673-0cf8ad43e917,eco-movement,2023-06-28T11:46:08.539000+00:00,7.762694,48.825613,NaN,NaN,False,False,False
2,ChargePoint,NaN,info@chargepoint.com,ChargePoint,info@chargepoint.com,NaN,ACU_Poste_De_Garde_Haguenau,ATHTBE1004019,ATHTBE1004019,ACU_Poste_De_Garde_Haguenau,...,b11113db-875d-41c7-8673-0cf8ad43e917,eco-movement,2023-06-28T11:46:08.539000+00:00,7.762694,48.825613,NaN,NaN,False,False,False
3,ChargePoint,NaN,info@chargepoint.com,ChargePoint,info@chargepoint.com,NaN,ACU_Poste_De_Garde_Haguenau,ATHTBE1004020,ATHTBE1004020,ACU_Poste_De_Garde_Haguenau,...,b11113db-875d-41c7-8673-0cf8ad43e917,eco-movement,2023-06-28T11:46:08.539000+00:00,7.762694,48.825613,NaN,NaN,False,False,False
4,ChargePoint,NaN,info@chargepoint.com,ChargePoint,info@chargepoint.com,NaN,HAG_P22_Slave_3,ATHTBE1004957,ATHTBE1004957,HAG_P22_Slave_3,...,b11113db-875d-41c7-8673-0cf8ad43e917,eco-movement,2023-06-28T11:46:08.539000+00:00,7.761841,48.827040,NaN,NaN,False,False,False


Cette table comporte 210129 lignes pour 52 variables.

In [3]:
list(df_irve.columns)

['nom_amenageur',
 'siren_amenageur',
 'contact_amenageur',
 'nom_operateur',
 'contact_operateur',
 'telephone_operateur',
 'nom_enseigne',
 'id_station_itinerance',
 'id_station_local',
 'nom_station',
 'implantation_station',
 'adresse_station',
 'code_insee_commune',
 'coordonneesXY',
 'nbre_pdc',
 'id_pdc_itinerance',
 'id_pdc_local',
 'puissance_nominale',
 'prise_type_ef',
 'prise_type_2',
 'prise_type_combo_ccs',
 'prise_type_chademo',
 'prise_type_autre',
 'gratuit',
 'paiement_acte',
 'paiement_cb',
 'paiement_autre',
 'tarification',
 'condition_acces',
 'reservation',
 'horaires',
 'accessibilite_pmr',
 'restriction_gabarit',
 'station_deux_roues',
 'raccordement',
 'num_pdl',
 'date_mise_en_service',
 'observations',
 'date_maj',
 'cable_t2_attache',
 'last_modified',
 'datagouv_dataset_id',
 'datagouv_resource_id',
 'datagouv_organization_or_owner',
 'created_at',
 'consolidated_longitude',
 'consolidated_latitude',
 'consolidated_code_postal',
 'consolidated_commune',
 '

Vérification du type des codes :

In [4]:
df_irve["code_insee_commune"].dtype

dtype('O')

dtype('O') = Object (texte)
La colonne `code_insee_commune` est stockée comme du texte.

In [5]:
df_irve["code_insee_commune"]

0           NaN
1           NaN
2           NaN
3           NaN
4           NaN
          ...  
210124    95572
210125    59359
210126      NaN
210127      NaN
210128      NaN
Name: code_insee_commune, Length: 210129, dtype: object

In [6]:
df_irve["code_insee_commune"].isna().sum()

np.int64(64848)

Vérification de la longueur des codes :

In [7]:
df_irve["code_insee_commune"].str.len().value_counts()

code_insee_commune
5.0    112779
Name: count, dtype: int64

Attention, le code géographique n'est pas renseigné pour 64848 lignes sur 210129.

## 2. Voitures particulières immatriculées par commune et par type de recharge

In [8]:
df_ve = pd.read_csv(
 'https://www.data.gouv.fr/api/1/datasets/r/90e0d717-deda-4bdc-9987-f82faac5bc93',
 encoding='latin-1'
)

print(f"Shape : {df_ve.shape}")
df_ve.head()

Shape : (703545, 8)


/tmp/ipykernel_36252/3535292153.py:1: DtypeWarning: Columns (0,2) have mixed types. Specify dtype option on import or set low_memory=False.
  df_ve = pd.read_csv(


,CODGEO,LIBGEO,EPCI,LIBEPCI,DATE_ARRETE,NB_VP_RECHARGEABLES_EL,NB_VP_RECHARGEABLES_GAZ,NB_VP
0,01022,ARTEMARE,200040350.0,CC Bugey Sud,2025-09-30,26,0,1446
1,01022,ARTEMARE,200040350.0,CC Bugey Sud,2025-06-30,27,0,1437
2,01022,ARTEMARE,200040350.0,CC Bugey Sud,2025-03-31,20,0,1425
3,01022,ARTEMARE,200040350.0,CC Bugey Sud,2024-12-31,20,0,1416
4,01022,ARTEMARE,200040350.0,CC Bugey Sud,2024-09-30,21,0,1406


In [9]:
list(df_ve.columns)

['CODGEO',
 'LIBGEO',
 'EPCI',
 'LIBEPCI',
 'DATE_ARRETE',
 'NB_VP_RECHARGEABLES_EL',
 'NB_VP_RECHARGEABLES_GAZ',
 'NB_VP']

Vérification du type des codes :

In [10]:
df_ve["CODGEO"].dtype

dtype('O')

In [11]:
df_ve["CODGEO"]

0         01022
1         01022
2         01022
3         01022
4         01022
          ...  
703540    97617
703541    97617
703542    97617
703543    97617
703544    97617
Name: CODGEO, Length: 703545, dtype: object

In [12]:
df_ve["CODGEO"].isna().sum()

np.int64(0)

Vérification de la longueur des codes :

In [13]:
df_ve["CODGEO"].str.len().value_counts()

CODGEO
5.0    393216
Name: count, dtype: int64

## 3. Revenu des français à la commune

In [14]:
df_revenus = pd.read_csv(
 'https://www.data.gouv.fr/api/1/datasets/r/516130bc-4dcb-47f5-8347-ae96553c43ab',sep=';'
)

print(f"Shape : {df_revenus.shape}")
df_revenus.head()

Shape : (34926, 57)


/tmp/ipykernel_36252/2507393620.py:1: DtypeWarning: Columns (1) have mixed types. Specify dtype option on import or set low_memory=False.
  df_revenus = pd.read_csv(


,Nom géographique GMS,Code géographique,Libellé géographique,[DISP] Nbre de ménages fiscaux,[DISP] Nbre de personnes dans les ménages fiscaux,[DISP] Nbre d'unités de consommation dans les ménages fiscaux,[DISP] 1ᵉʳ quartile (€),[DISP] Médiane (€),[DISP] 3ᵉ quartile (€),[DISP] Écart interquartile (€),...,[DEC] 9ᵉ décile (€),[DEC] Rapport interdécile D9/D1,[DEC] S80/S20,[DEC] Iice de Gini,[DEC] Part des revenus d’activité (%),[DEC] dont part des salaires et traitements (%),[DEC] dont part des iemnités de chômage (%),[DEC] dont part des revenus des activités non salariées (%),"[DEC] Part des pensions, retraites et rentes (%)",[DEC] Part des autres revenus (%)
0,L'Abergement-Clémenciat,01001,L'Abergement-Clémenciat,346.0,895.0,590.8,NaN,25820.0,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
1,L'Abergement-de-Varey,01002,L'Abergement-de-Varey,115.0,266.0,181.0,NaN,24480.0,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
2,Ambérieu-en-Bugey,01004,Ambérieu-en-Bugey,6855.0,15092.0,10398.2,15800.0,21660.0,28430.0,12630.0,...,39380.0,"5,6",8.1,0.368,67.9,61.5,3.2,3.2,24.4,7.7
3,Ambérieux-en-Dombes,01005,Ambérieux-en-Dombes,800.0,2028.0,1329.7,20010.0,24610.0,31180.0,11170.0,...,43610.0,"3,2",4.4,0.298,73.2,67.2,2.2,3.8,19.7,7.1
4,Ambléon,01006,Ambléon,51.0,107.0,76.6,NaN,24210.0,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN


In [15]:
list(df_revenus.columns)

['Nom géographique GMS',
 'Code géographique',
 'Libellé géographique',
 '[DISP] Nbre de ménages fiscaux',
 '[DISP] Nbre de personnes dans les ménages fiscaux',
 "[DISP] Nbre d'unités de consommation dans les ménages fiscaux",
 '[DISP] 1ᵉʳ quartile (€)',
 '[DISP] Médiane (€)',
 '[DISP] 3ᵉ quartile (€)',
 '[DISP] Écart interquartile (€)',
 '[DISP] 1ᵉʳ décile (€)',
 '[DISP] 2ᵉ décile (€)',
 '[DISP]3ᵉ décile (€)',
 '[DISP] 4ᵉ décile (€)',
 '[DISP] 6ᵉ décile (€)',
 '[DISP] 7ᵉ décile (€)',
 '[DISP] 8ᵉ décile (€)',
 '[DISP] 9ᵉ décile (€)',
 '[DISP] Rapport interdécile 9ᵉ décile/1ᵉʳ decile',
 '[DISP] S80/20',
 '[DISP] Iice de Gini',
 '[DISP] Part des revenus d’activité (%)',
 '[DISP] dont part des salaires et traitements(%)',
 '[DISP] dont part des iemnités de chômage (%)',
 '[DISP] dont part des revenus des activités non salariées (%)',
 '[DISP] Part des pensions, retraites et rentes (%)',
 '[DISP] Part des revenus du patrimoine et autres revenus (%)',
 "[DISP] Part de l'ensemble des prestat

Vérification du type des codes :

In [16]:
df_revenus["Code géographique"].dtype

dtype('O')

In [17]:
df_revenus["Code géographique"]

0        01001
1        01002
2        01004
3        01005
4        01006
         ...  
34921    97420
34922    97421
34923    97422
34924    97423
34925    97424
Name: Code géographique, Length: 34926, dtype: object

In [18]:
df_revenus["Code géographique"].isna().sum()

np.int64(0)

Vérification de la longueur des codes :

In [19]:
df_revenus["Code géographique"].str.len().value_counts()

Code géographique
5.0    16384
Name: count, dtype: int64

Regardons si certaines valeurs sont présentes dans certains datasets masi absentes dans d'autres.

In [20]:
len(set(df_irve["code_insee_commune"]))

27624

In [21]:
len(set(df_ve["CODGEO"]))

35206

In [22]:
len(set(df_revenus["Code géographique"]))

34926

In [ ]:
df_ve["CODGEO"] = df_ve["CODGEO"].astype(str).str.zfill(5)
df_revenus["Code géographique"] = df_revenus["Code géographique"].astype(str).str.zfill(5)
df_irve["code_insee_commune"] = df_irve["code_insee_commune"].astype(str).str.zfill(5)

In [31]:
len(set(df_ve["CODGEO"]) - set(df_revenus["Code géographique"]))

275

In [32]:
len(set(df_ve["CODGEO"]) - set(df_irve["code_insee_commune"]))

29417

In [33]:
len(set(df_revenus["Code géographique"]) - set(df_irve["code_insee_commune"]))

29153

In [34]:
len(set(df_revenus["Code géographique"]) - set(df_ve["CODGEO"]))

4

In [35]:
len(set(df_irve["code_insee_commune"]) - set(df_revenus["Code géographique"]))

5201

In [36]:
len(set(df_irve["code_insee_commune"]) - set(df_ve["CODGEO"]))

5194

Il y a anormalement beaucoup de différences entre df_irve["code_insee_commune"] et df_ve["CODGEO"] et
df_revenus["Code géographique"].

uniformiser les clés géographique = code_geo (regarder si c'est pas déjà ok)